### PR TITLE
ASM2D documentation

### LINKED ISSUE #597: ASM2D Documentation
Documentation needed for ASM2D model. Can emulate ASM1D documentation as needed to initiate

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
Add documentation for activated sludge model no. 2D (ASM2D).

## Changes proposed in this PR:
- add .rst file
- slight revision of asm1 rst file and doc strings in asm2d model
-  todo notes in ASM2d identifying potential errors in 2 process rate equations, which could explain deviations previously identified by @andrewlee94 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/property_models/ASM2D.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
+ASM2D Property Package
+======================
+
+This package implements properties and reactions of an activated sludge model for biological nutrient removal from wastewater using an activated sludge biological reactor with biological phosphorus removal as provided in Henze, M. et al. (1999) [1].
+
+This Activated Sludge Model no.2D (ASM2D) property/reaction package:
+   * supports 'H2O', 'S_A', 'S_F', 'S_I', S_N2, S_NH4, S_NO3, S_O2, S_PO4, S_ALK, X_AUT, X_H, X_I, X_MeOH, X_MeP, X_PAO, X_PHA, X_PP, X_S, and X_TSS as components
+   * supports only liquid phase
+
+Limitations of the model, based on the original literature referenced, are as follows:
```
Comment (adam-a-a) [inline docs/technical_reference/property_models/ASM2D.rst]:
```suggestion
Limitations of the model, as noted in [1], are as follows:
```

**Comment by adam-a-a** [comment]
@yalinli2 I think I may have found 2 potential errors in our ASM2D model, but I'm uncertain whether @andrewlee94 intentionally deviated from the literature because he regarded the original rate equations as being erroneous, or if it was an oversight. Specifically, check out rate equations 6 and 7; one of the denominators, according to the literature, should be the sum of S_ALK and S_ALK, while our model uses the sum of S_ALK and K_ALK. What are your thoughts on this?

**Comment by yalinli2** [comment]
Hey Adam I'm not exactly sure if I understood you correctly, but I think Andrew is correct? E.g., for R6, the literature indicates S<sub>ALK</sub>/(K<sub>ALK</sub>+S<sub>ALK</sub>), and here is what Andrew has:
https://github.com/watertap-org/watertap/blob/85b2599cb82832da6f65c239844c00163fd80760/watertap/property_models/activated_sludge/asm2d_reactions.py#L1226

I think `b.params.K_ALK` is  K<sub>ALK</sub> and `b.state_ref.alkalinity` is S<sub>ALK</sub>? Did I misunderstood?


> @yalinli2 I think I may have found 2 potential errors in our ASM2D model, but I'm uncertain whether @andrewlee94 intentionally deviated from the literature because he regarded the original rate equations as being erroneous, or if it was an oversight. Specifically, check out rate equations 6 and 7; one of the denominators, according to the literature, should be the sum of S_ALK and S_ALK, while our model uses the sum of S_ALK and K_ALK. What are your thoughts on this?

**Comment by adam-a-a** [comment]
> Hey Adam I'm not exactly sure if I understood you correctly, but I think Andrew is correct? E.g., for R6, the literature indicates SALK/(KALK+SALK), and here is what Andrew has:
> 
> https://github.com/watertap-org/watertap/blob/85b2599cb82832da6f65c239844c00163fd80760/watertap/property_models/activated_sludge/asm2d_reactions.py#L1226
> 
> I think `b.params.K_ALK` is KALK and `b.state_ref.alkalinity` is SALK? Did I misunderstood?
> 
> > @yalinli2 I think I may have found 2 potential errors in our ASM2D model, but I'm uncertain whether @andrewlee94 intentionally deviated from the literature because he regarded the original rate equations as being erroneous, or if it was an oversight. Specifically, check out rate equations 6 and 7; one of the denominators, according to the literature, should be the sum of S_ALK and S_ALK, while our model uses the sum of S_ALK and K_ALK. What are your thoughts on this?

Looking at Table 7 in Henze et al: 
image

**Comment by adam-a-a** [comment]
> > Hey Adam I'm not exactly sure if I understood you correctly, but I think Andrew is correct? E.g., for R6, the literature indicates SALK/(KALK+SALK), and here is what Andrew has:
> > https://github.com/watertap-org/watertap/blob/85b2599cb82832da6f65c239844c00163fd80760/watertap/property_models/activated_sludge/asm2d_reactions.py#L1226
> > 
> > I think `b.params.K_ALK` is KALK and `b.state_ref.alkalinity` is SALK? Did I misunderstood?
> > > @yalinli2 I think I may have found 2 potential errors in our ASM2D model, but I'm uncertain whether @andrewlee94 intentionally deviated from the literature because he regarded the original rate equations as being erroneous, or if it was an oversight. Specifically, check out rate equations 6 and 7; one of the denominators, according to the literature, should be the sum of S_ALK and S_ALK, while our model uses the sum of S_ALK and K_ALK. What are your thoughts on this?
> 
> Looking at Table 7 in Henze et al: image

While all other rate equations that include these terms divide by the sum of S_alk and K_alk, those two equations appear to be the exception. So I wasn't sure if this was an error in our model or in the literature. Is there an alternative paper that lists all the process rate equations for ASM2D that we can compare against?

**Comment by yalinli2** [comment]
Oh boy which version were you looking at? I think you probably spotted a typo! I'm using this [2015 book](https://iwaponline.com/ebooks/book/96/Activated-Sludge-Models-ASM1-ASM2-ASM2d-and-ASM3) (also from Henze) and here is Table 3.7

Screen Shot 2022-07-06 at 10 31 02 AM

> Looking at Table 7 in Henze et al:

**Comment by adam-a-a** [comment]
> Oh boy which version were you looking at? I think you probably spotted a typo! I'm using this [2015 book](https://iwaponline.com/ebooks/book/96/Activated-Sludge-Models-ASM1-ASM2-ASM2d-and-ASM3) (also from Henze) and here is Table 3.7
> 
> Screen Shot 2022-07-06 at 10 31 02 AM
> 
> > Looking at Table 7 in Henze et al:

I was looking at the original paper by Henze et al. (https://www.sciencedirect.com/science/article/pii/S0273122398008294). Wow, so I guess the original work had typos!

**Review by yalinli2 — APPROVED** [review]

**Review by tarnold17 — APPROVED** [review]

**Pull request merged**